client/my-sites/hosting/staging-site-card/confirm-modal-button.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button } from '@automattic/components';
+import styled from '@emotion/styled';
+import { Modal } from '@wordpress/components';
+import { useState } from 'react';
+
+const ActionButtons = styled.div( {
+	display: 'flex',
+	gap: '1em',
+} );
+
+type ConfirmationModalButtonProps = {
+	onConfirm?: () => void;
+	onCancel?: () => void;
+	isBusy?: boolean;
+	isPrimary?: boolean;
+	isScary?: boolean;
+	isBorderless?: boolean;
+	isPlain?: boolean;
+	isTransparent?: boolean;
+	isCompact?: boolean;
+	disabled?: boolean;
+	children: React.ReactNode;
+	modalTitle: string;
+	modalMessage: string;
+	confirmLabel: string;
+	cancelLabel: string;
+};
+
+export function ConfirmationModalButton( {
```

Comment:
The component is not a button. Could `ConfirmationModal` be a better fit? Also, it would be good to keep the file name consistent with the main export name.